processor/logdedupprocessor/processor_test.go

```diff
@@ -109,13 +119,15 @@ func TestShutdownBeforeStart(t *testing.T) {
 		LogCountAttribute: defaultLogCountAttribute,
 		Interval:          1 * time.Second,
 		Timezone:          defaultTimezone,
+		Conditions:        []string{},
 		ExcludeFields: []string{
 			fmt.Sprintf("%s.remove_me", attributeField),
 		},
 	}
 
 	// Create a processor
 	p, err := newProcessor(cfg, logsSink, settings)
+	p.conditions = nil
```

Comment:
Having to directly manipulate the internals of unexported structures in order to set up the test is a sign that the test setup is fragile and/or unrepresentative. Testing against the exported API only should avoid this. Can we construct a config and pass it to a factory?